### PR TITLE
BugFix: Flow run logs flashing because of polling and subscription reactivity issues

### PR DESCRIPTION
**Pull request by pleek91**
# Description
The flow run logs use the flow run id in its filter. Since the flow run itself is on a 5 second interval, reactive effects are triggered every 5 seconds. Since the `usePaginatedSubscription ` composition joins multiple calls together, the reactive effect of the filter empties the response array. 

There used to be a check in place that checked for equality of the watcher's new and old values. But the old and new values [will be the same when deep watching](https://vuejs.org/guide/essentials/watchers.html#deep-watchers) an object. So that [check was removed](https://github.com/PrefectHQ/prefect-ui-library/commit/85afa18c369b5a4bdc86d29d56c12b3f8aaccab5) in order to support the latest filters work. Removing that ensured filter objects were accurate, but also caused paginated subscriptions to be be resubscribed even when the actual value itself hadn't changed. 

Putting back an equality check that actually works for both simple and object values in the form of a new utility called `uniqueValueWatcher`. Which is effectively the following but written in a way that supports complex objects an arrays as the watch source. 

```typescript
watch(value, (newValue, oldValue) => {
  if(newValue === oldValue) {
    return
  }
  // do stuff
})
```

### CODE REVIEW

**Comment by zhen0** [comment]
Thanks for the speedy turnaround on this @pleek91. I can't check this locally right now. We'll try get this approved this evening but if we can't are you ok if we merge this (after approval) in your absence tomorrow?

File: src/utilities/reactivity.ts
Comment (stackoverfloweth) [inline src/utilities/reactivity.ts:31]: love this abstraction. Might be worth a composition someday in vue-compositions? `useUniqueValueWatcher`?

**Review by stackoverfloweth — APPROVED** [review]

File: src/utilities/reactivity.ts
Comment (pleek91) [reply src/utilities/reactivity.ts:31]: I thought about that, but its a wrapper around `watch` which isn't a composition. Currently vue-compositions doesn't export any global utilities. So this seemed like the best place for now.

**Comment by pleek91** [comment]
@zhen0 I'm going to get this out tonight so no one is blocked by it

**Comment by zhen0** [comment]
> @zhen0 I'm going to get this out tonight so no one is blocked by it

Fantastic. Thank you!

**Pull request merged**
